Gemfile

```diff
@@ -56,26 +56,22 @@ group :backend do
 end
 
 group :admin do
-  gem 'solidus_admin', path: 'admin', require: false
   gem 'tailwindcss-rails', '~> 3.0', require: false
+end
+
+group :admin, :legacy_promotions, :promotions do
+  gem 'solidus_admin', path: 'admin', require: false
   gem 'axe-core-rspec', '~> 4.8', require: false
   gem 'axe-core-capybara', '~> 4.8', require: false
 end
 
-group :legacy_promotions do
+group :legacy_promotions, :promotions do
```

Comment:
How do we handle backward compatibily when we will remove legacy_promotions from the system? I am thinking about detaching those gems by removing all the code that reference to legacy_promotions from promotions in a later version (eg v4.5). This means that if a store wants to migrate, it has to make it before that version is released. Did you have some other ideas?

Reply:
I would first make `solidus_promotions` the new default (Solidus 5) (by having `solidus` depend on it), still keep `solidus_legacy_promotions` until Solidus 6 comes up. Solidus 6 then moves `legacy_promotions` to its own repo, and `promotions` can shed the dependency. I think it's smart to play this as a long game.

Reply:
I like the plan. There's no hurry in getting rid of the legacy_promotions gem.